datadog_checks_base/datadog_checks/base/utils/profiling.py

```diff
@@ -0,0 +1,35 @@
+# (C) Datadog, Inc. 2018-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from ddtrace.profiling import Profiler
+
+
+class Profiling(object):
+    _instance = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(Profiling, cls).__new__(cls)
+            cls._instance._running = False
+            cls._instance._profiler = None
+
+        return cls._instance
+
+    # TODO: Double check if we need to use any concurrency control mechanism
```

Comment:
I think you do. By default we have 4 checks in parallel and some customers bump that number

Reply:
Thanks for the confirmation. I would add some concurrency control 😄

Reply:
@FlorentClarret certainly it has to be done when the GIL will be abandon but is the lock still needed now since two threads in my poor Python comprehension could not execute concurrently in the same process/interpreter?

Reply:
@iglendd You are totally right, this is documented in this [PEP](https://peps.python.org/pep-0703/?adobe_mc=MCMID%3D73102239162756344273860943530194341740%7CMCORGID%3DA8833BC75245AF9E0A490D4D%2540AdobeOrg%7CTS%3D1715949371). This will also be opt-in in Python 3.13, so we would need to explicitly enable it. 

You could go with a simple `# TODO` here for now, hoping that the one that will enable it one day (if we enable it) will go through all the TODOs we have in the code 😅